exla/mix.exs

```diff
@@ -150,14 +167,21 @@ defmodule EXLA.MixProject do
     cached_so = Path.join([xla_cache_dir(), "exla", cache_key, "libexla.so"])
     cached? = File.exists?(cached_so)
```

Comment:
I think you can mostly remove the changes to this function and instead simply write this:

```suggestion
    cached? = System.get_env("XLA_FORCE_REBUILD") == nil and File.exists?(cached_so)
```

Reply:
This will be enough to force a rebuild and write the new build to cache.

Reply:
@josevalim I changed to using env vars, but kept 2 separate ones because there are many instances in which I wanted to leverage the `.o` caching while getting rid of the libexla.so file and cache, as well as instances where I wanted to clean-slate everything.

Reply:
My problem with the above is that, if something is not working, I need to set two env vars to clear everything. So why not a single env var with different values?

  * `XLA_FORCE_REBUILD=1` - remove all
  * `XLA_FORCE_REBUILD=partial` - keep .o

Also, is there any situation where you don't want to cache files to be rewritten? I don't see a reason to skip that as part of these options?

Finally note that we use `XLA` env vars for pretty much everything, since we have `XLA_CACHE_DIR`, etc.

Reply:
I'm ok with having special values for the env var and having a single one.

I disagree with using `XLA` as the prefix, though, as this doesn't have anything to do with `:xla`. `XLA_CACHE_DIR` being used in EXLA is just an artifact of "let's use the same subdir for both XLA and EXLA". The origin of the variable is in `:xla`.

> Also, is there any situation where you don't want to cache files to be rewritten? I don't see a reason to skip that as part of these options?

As for this part, I don't think we are skipping the rewrite.

Reply:
I refactored the `cached?` check so that there's less confusion about whether rewrites will happen.
`cached?` will now be set to false regardless of the file existing if force rebuild is set.